azure/scope/machinepoolmachine.go

```diff
@@ -274,14 +287,21 @@ func (s *MachinePoolMachineScope) UpdateStatus(ctx context.Context) error {
 		node    *corev1.Node
 		err     error
 	)
-	if nodeRef == nil || nodeRef.Name == "" {
+
+	if s.ProviderID() == "" {
```

Comment:
@CecileRobertMichon I made some changes to the if logic based on your comment. Is this what you were describing?

Reply:
yep looks good